src/main/java/de/retest/web/DefaultValuesProvider.java

```diff
@@ -0,0 +1,95 @@
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import de.retest.ui.DefaultValueFinder;
+import de.retest.ui.descriptors.IdentifyingAttributes;
+
+public class DefaultValuesProvider implements DefaultValueFinder {
+
+	public static final class ReadDefaultValuesException extends RuntimeException {
+
+		private static final long serialVersionUID = 1L;
+
+		public ReadDefaultValuesException( final Exception e ) {
+			super( "Could not load default CSS values from " + DEFAULTS_FILE_PATH, e );
+		}
+	}
+
+	public static final String DEFAULTS_FILE_PATH = "/defaults.yaml";
+
+	private final Map<String, Map<String, String>> defaultValues;
+
+	public DefaultValuesProvider() {
+		try ( final InputStream url = getClass().getResourceAsStream( DEFAULTS_FILE_PATH ) ) {
+			defaultValues = readAttributesConfigFromFile( url );
+		} catch ( final Exception e ) {
+			throw new ReadDefaultValuesException( e );
+		}
+	}
+
+	private Map<String, Map<String, String>> readAttributesConfigFromFile( final InputStream in ) throws IOException {
+		final Map<String, Map<String, String>> defaultValues = new HashMap<>();
+		final ObjectMapper mapper = new ObjectMapper( new YAMLFactory() );
+		final JsonNode jsonNode = mapper.readTree( in );
+		for ( final Iterator<Entry<String, JsonNode>> elements = jsonNode.fields(); elements.hasNext(); ) {
+			final Entry<String, JsonNode> field = elements.next();
+			final Map<String, String> defaults = new HashMap<>();
+			final ArrayNode valuesNode = (ArrayNode) field.getValue();
+			for ( final Iterator<JsonNode> values = valuesNode.elements(); values.hasNext(); ) {
+				final Entry<String, JsonNode> value = values.next().fields().next();
+				defaults.put( value.getKey(), value.getValue().asText() );
+			}
+			defaultValues.put( field.getKey(), defaults );
+		}
+		return defaultValues;
+	}
+
+	public String getDefaultValue( final String tag, final String attribute ) {
+		final Map<String, String> defaults = defaultValues.get( tag.toLowerCase() );
+		if ( defaults != null ) {
+			final String defaultValue = defaults.get( attribute.toLowerCase() );
+			if ( defaultValue != null ) {
+				return defaultValue;
+			}
+		}
+		return defaultValues.get( "all" ).get( attribute.toLowerCase() );
+	}
+
+	@Override
+	public Serializable getDefaultValue( final IdentifyingAttributes comp, final String attributesKey ) {
+		return getDefaultValue( comp.getType(), attributesKey );
+	}
+
+	public boolean isDefault( final String tag, final String attribute, final String attributeValue ) {
+		final String defaultValue = getDefaultValue( tag, attribute );
+		if ( defaultValue != null ) {
+			return defaultValue.equalsIgnoreCase( attributeValue );
+		}
+		if ( attributeValue == null || attributeValue.trim().isEmpty() ) {
+			return true;
+		}
+		if ( attributeValue.equals( "auto" ) ) {
+			return true;
+		}
+		if ( attributeValue.equals( "normal" ) ) {
+			return true;
+		}
+		if ( attributeValue.equals( "0px" ) ) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=52&issues=AWURSaL-yxJLWg29qApz&open=AWURSaL-yxJLWg29qApz)

[ISSUE_KEY]: AWURSaL-yxJLWg29qApz